sympy/functions/special/beta_functions.py

```diff
@@ -100,7 +108,9 @@ def fdiff(self, argindex):
             raise ArgumentIndexError(self, argindex)
 
     @classmethod
-    def eval(cls, x, y):
+    def eval(cls, x, y=None):
+        if y == None:
+            return beta(x, x)
```

Comment:
Test for `None` with `is`. Also it's more efficient to do:
```
if y is None:
    y = x
```
rather than the recursive call.

Reply:
Sure, I'll change that

Reply:
If I don't do the recursive call, it outputs an object that looks like a single argument object. Eg- 
``` python
>>> from sympy import beta
>>> x, y = symbols('x y')
>>> beta(x, y)
beta(x, y)
>>> beta(x)
beta(x) #Was previously beta(x, x)
```
So, which is better?

Reply:
I guess the recursive call is fine then

Reply:
Ok